packages/app-runtime/src/modules/runtimeEvents/IdentityDeletionProcessStatusChangedModule.ts

```diff
@@ -19,25 +21,44 @@ export class IdentityDeletionProcessStatusChangedModule extends AppRuntimeModule
     private async handleIdentityDeletionProcessStatusChanged(event: IdentityDeletionProcessStatusChangedEvent) {
         const identityDeletionProcess = event.data;
 
+        if (!identityDeletionProcess) {
+            const services = await this.runtime.getServices(event.eventTargetAddress);
+            const identityDeletionProcesses = await services.transportServices.identityDeletionProcesses.getIdentityDeletionProcesses();
+
+            const approvedIdentityDeletionProcess = identityDeletionProcesses.value.filter((idp) => idp.status === IdentityDeletionProcessStatus.Approved).at(0);
+            const deletionDate = approvedIdentityDeletionProcess?.gracePeriodEndsAt ? CoreDate.from(approvedIdentityDeletionProcess.gracePeriodEndsAt) : undefined;
+
+            await this.updateLocalAccountDeletionDate(event.eventTargetAddress, deletionDate, true);
+            return;
+        }
+
         switch (identityDeletionProcess.status) {
             case IdentityDeletionProcessStatus.Approved:
-                await this.runtime.multiAccountController.updateLocalAccountDeletionDate(event.eventTargetAddress, CoreDate.from(identityDeletionProcess.gracePeriodEndsAt!));
+                await this.updateLocalAccountDeletionDate(event.eventTargetAddress, CoreDate.from(identityDeletionProcess.gracePeriodEndsAt!));
                 break;
 
             case IdentityDeletionProcessStatus.Cancelled:
-                const account = await this.runtime.multiAccountController.getAccountByAddress(event.eventTargetAddress);
-                const previousDeletionDate = account.deletionDate;
-
-                if (!previousDeletionDate) break;
-
-                await this.runtime.multiAccountController.updateLocalAccountDeletionDate(event.eventTargetAddress, undefined);
+                await this.updateLocalAccountDeletionDate(event.eventTargetAddress, undefined);
                 break;
 
             default:
                 break;
         }
     }
 
+    private async updateLocalAccountDeletionDate(eventTargetAddress: string, deletionDate: CoreDate | undefined, publishEvent = false) {
```

Comment:
Would it be cleaner to adjust the name of the function such that it doesn't exactly match the name of the MultiAccountController functions, since we do more stuff here?

Reply:
I don't see a reason why matching function names are a problem.